porcupine/settings.py

```diff
@@ -201,12 +219,18 @@ def set(self, option_name: str, value: object, *, from_config: bool = False) ->
         """
         # ...even though this method isn't named 'set_'. But the docstring is
         # used in settings.rst to document a global "function".
+
+        if already_converted and not from_config:
+            raise RuntimeError("already_converted=True can't be used without from_config=True")
+
         if option_name not in self._options and from_config:
-            self._unknown_options[option_name] = value
+            self._unknown_options[option_name] = _UnknownOption(
+                value, converter_called=already_converted
+            )
             return
 
         option = self._options[option_name]
-        if from_config:
+        if from_config and not already_converted:
```

Comment:
Should refactor it so that this line says `if not already_converted`. The `already_converted` boolean only makes sense with `from_config` should change so it is always good